bpf/lib/nat.h

```diff
@@ -884,7 +884,7 @@ snat_v4_nat(struct __ctx_buff *ctx, const struct ipv4_nat_target *target, __s8 *
 			return DROP_INVALID;
 		if (icmphdr.type != ICMP_ECHO &&
 		    icmphdr.type != ICMP_ECHOREPLY)
-			return DROP_NAT_UNSUPP_PROTO;
+			return NAT_PUNT_TO_STACK;
```

Comment:
Can we please limit this PR to the revSNAT path? That's where we want to improve the error handling ... otherwise this is just getting too confusing.

Having said that, I believe this change (in the SNAT path) would be great as a follow-on and make other users happy, eg. https://github.com/cilium/cilium/issues/23955 or https://github.com/cilium/cilium/issues/15890